From: Chad Monroe <chad@monroe.io>
Date: Thu, 10 Apr 2025 10:28:40 -0700
Subject: [PATCH] net: ethernet: mediatek: don't set queue speeds for dsa1 

currently the only device with netdev dsa1 connected to GMAC2 is the
SDG-9000. due to broken flow offload, traffic doesn't get assigned to the
correct QDMA queue. most traffic uses QDMA tx queue 3.. without this
patch, queue 3 could be limited by the link speed of LAN1.

once we switch to the upstream Maxlinear DSA driver this hack should
be removed.

Signed-off-by: Chad Monroe <chad@monroe.io>
---
 drivers/net/ethernet/mediatek/mtk_eth_soc.c |    8 ++++++++
 1 file changed, 8 insertions(+)

--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -3551,6 +3551,14 @@ static int mtk_device_event(struct notif
 	if (event != NETDEV_CHANGE)
 		return NOTIFY_DONE;
 
+	switch (mac->id) {
+	case MTK_GMAC2_ID:
+		if (strcmp(netdev_name(dev), "dsa1"))
+			return NOTIFY_DONE;
+	default:
+		break;
+	}
+
 	netdev_for_each_lower_dev(dev, ldev, iter) {
 		if (netdev_priv(ldev) == mac)
 			goto found;
